packages/components/src/templates/next/components/complex/CollectionBlock/utils/getCollectionParent.ts

```diff
@@ -0,0 +1,40 @@
+import type { IsomerSitemap, IsomerSiteProps } from "~/types"
+import type { IsomerCollectionPageSitemap } from "~/types/sitemap"
+
+const isCollectionParent = (
+  node: IsomerSitemap,
+  collectionId: string,
+): node is IsomerCollectionPageSitemap => {
+  return node.id === collectionId && node.layout === "collection"
+}
+
+interface GetCollectionParentProps {
+  site: IsomerSiteProps
+  collectionId: string
+}
+
+export const getCollectionParent = ({
+  site,
+  collectionId,
+}: GetCollectionParentProps): IsomerCollectionPageSitemap => {
```

Comment:
question: why not `getSitemapAsArray` then find the node from the array?

Reply:
updated in https://github.com/opengovsg/isomer/pull/1250/commits/b130d7de96f59669511e9622d489bb75c45ffac3 - didn't occur to me that we do that. makes a lot of sense esp. since we using it for generating collection aldy. thanks!